src/MiniHallway.jl

```diff
@@ -0,0 +1,88 @@
+# Mini Hallway problem defined in http://cs.brown.edu/research/ai/pomdp/examples/mini-hall2.POMDP.
+# Original idea published in Littman, Cassandra and Kaelbling's ML-95 paper.
+
+# Basic parameters are:
+# discount: 0.950000
+# values: reward
+# states: 13
+# actions: 3
+# observations: 9
+# The rest is available at link
+
+
+struct Observation
+    no::Int
+end
+
+struct MiniHallway <: POMDP{Int, Int, Observation}
+    T::Array{SparseCat, 2}
```

Comment:
What is the reason for putting these in a matrix like this? I think it would be more readable to put it in a `transition` function, for instance
```julia
function transition(m::MiniHallway, s, a)
    if a == 2
        return Deterministic(s+1)
    # ... I can't discern the pattern for the rest of them, but I'm sure there is one
end
```

Reply:
Same for `O` and `R`

Reply:
I could not find a pattern for action 1. That could be why I did not do this already.
I created functions for actions 2 and 3 and kept the vector for action 1.

Also created functions for `O` and `R`.